pkg/webservice/handlers.go

```diff
@@ -521,3 +522,49 @@ func updateConfiguration(conf string) (string, error) {
 	}
 	return "", fmt.Errorf("config plugin not found")
 }
+
+func getQueueApplications(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	writeHeaders(w)
+	if len(vars) != 2 {
+		http.Error(w, "Mandatory parameters are missing in URL path. Please check the usage documentation", http.StatusBadRequest)
+		return
+	}
+	var queueName = vars["queue"]
+	if len(queueName) == 0 {
+		http.Error(w, "Queue is missing in URL path. Please check the usage documentation", http.StatusBadRequest)
+		return
+	}
+	queueErr := validateQueue(queueName)
+	if queueErr != nil {
+		http.Error(w, queueErr.Error(), http.StatusBadRequest)
+		return
+	}
+	var partitionExists = false
+	var queueExists = false
+	var appsDao []*dao.ApplicationDAOInfo
+	lists := schedulerContext.GetPartitionMapClone()
+	for _, partition := range lists {
+		if partition.Name == vars["partition"] {
+			partitionExists = true
+			appList := partition.GetApplications()
```

Comment:
Is it safe to loop application here?

Reply:
I think so as it is in handler side unlike variables in core objects/classes.

Reply:
@wilfred-s cud u please share your thought on this comment?

Reply:
This function is supposed to get all applications from a queue, why we need to loop the partition?
Can we leverage `queue#getCopyOfApps()`?